app/models/member.rb

```diff
@@ -70,6 +70,10 @@ def deletable?
     member_roles.detect(&:inherited_from).nil?
   end
 
+  def deletable_role?(role)
+    member_roles.filter { _1.role_id == role.id }.none?(&:inherited_from)
```

Comment:
I can see that this doesn't make a lot of difference in the roles table where all member roles are loaded anyway, but I'd prefer a query like the following to avoid doing the filter in ruby.

```suggestion
    member_roles
      .where(role_id: role.id)
      .where.not(inherited_from: nil)
      .empty?
```

Reply:
Good point. Done in b6e4dbcbcc.

I introduced a scope for better readibility.